pageserver/src/tenant/ephemeral_file.rs

```diff
@@ -378,35 +430,33 @@ mod tests {
                 .await
                 .unwrap();
 
-        let cap = file.buffered_writer.inspect_buffer().capacity();
+        // mutable buffer and maybe_flushed buffer each has cap.
+        let cap = file.buffered_writer.inspect_mutable().capacity();
 
         let content: Vec<u8> = rand::thread_rng()
             .sample_iter(rand::distributions::Standard)
-            .take(cap + cap / 2)
+            .take(cap * 2 + cap / 2)
             .collect();
 
         file.write_raw(&content, &ctx).await.unwrap();
 
         // assert the state is as this test expects it to be
         assert_eq!(
             &file.load_to_io_buf(&ctx).await.unwrap(),
-            &content[0..cap + cap / 2]
+            &content[0..cap * 2 + cap / 2]
+        );
+        let md = file.buffered_writer.as_inner().path().metadata().unwrap();
+        assert!(
+            md.len() == cap.into_u64() || md.len() == 2 * cap.into_u64(),
+            "buffered writer requires one write to be flushed if we write 2.5x buffer capacity"
         );
```

Comment:
Same issue as in the earlier comment.

NB: the `load_to_io_buf` _above_ this assertion must _not_ require waiting, it should work at all times, even with background flushes ongoing.

Reply:
5acc61bdbcda38c93e6825da91a18706d3f1a7f3